pkg/apis/agones/v1/gameserver_test.go

```diff
@@ -505,8 +577,365 @@ func TestGameServerValidate(t *testing.T) {
+	causes, ok = gs.Validate()
+	for _, f := range causes {
+		fields = append(fields, f.Field)
+	}
+	assert.False(t, ok)
+	if assert.Len(t, causes, 1) {
+		assert.Equal(t, causes[0].Message, "Request must be less than or equal to memory limit")
+	}
+	assert.Contains(t, fields, "container")
+
+	// Memory negative request
+	gs = GameServer{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "dev-game",
+			Namespace: "default",
+		},
+		Spec: GameServerSpec{
+			Ports: []GameServerPort{{Name: "main",
+				ContainerPort: 7777,
+				PortPolicy:    Dynamic,
+			}},
+			Container: "testing",
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{Containers: []corev1.Container{
+					{
+						Name:  "testing",
+						Image: "testing/image",
+						Resources: corev1.ResourceRequirements{
+							Requests: corev1.ResourceList{
+								corev1.ResourceCPU:    resource.MustParse("30m"),
+								corev1.ResourceMemory: resource.MustParse("-32Mi"),
+							},
+							Limits: corev1.ResourceList{
+								corev1.ResourceCPU:    resource.MustParse("30m"),
+								corev1.ResourceMemory: resource.MustParse("32Mi"),
+							},
+						},
+					},
+				}}},
+		},
+	}
+	gs.ApplyDefaults()
+	causes, ok = gs.Validate()
+	for _, f := range causes {
+		fields = append(fields, f.Field)
+	}
+	assert.False(t, ok)
+	if assert.Len(t, causes, 1) {
+		assert.Equal(t, causes[0].Message, "Resource memory request value must be non negative")
+	}
+	assert.Contains(t, fields, "container")
+
+	// Memory negative limit
+	gs = GameServer{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "dev-game",
+			Namespace: "default",
+		},
+		Spec: GameServerSpec{
+			Ports: []GameServerPort{{Name: "main",
+				ContainerPort: 7777,
+				PortPolicy:    Dynamic,
+			}},
+			Container: "testing",
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{Containers: []corev1.Container{
+					{
+						Name:  "testing",
+						Image: "testing/image",
+						Resources: corev1.ResourceRequirements{
+							Requests: corev1.ResourceList{
+								corev1.ResourceCPU:    resource.MustParse("30m"),
+								corev1.ResourceMemory: resource.MustParse("32Mi"),
+							},
+							Limits: corev1.ResourceList{
+								corev1.ResourceCPU:    resource.MustParse("30m"),
+								corev1.ResourceMemory: resource.MustParse("-32Mi"),
+							},
+						},
+					},
+				}}},
+		},
+	}
+	gs.ApplyDefaults()
+	causes, ok = gs.Validate()
+	for _, f := range causes {
+		fields = append(fields, f.Field)
+	}
+	assert.False(t, ok)
+	if assert.Len(t, causes, 2) {
+		assert.Equal(t, causes[1].Message, "Resource memory limit value must be non negative")
+	}
+	assert.Contains(t, fields, "container")
+	runtime.FeatureTestMutex.Lock()
+	defer runtime.FeatureTestMutex.Unlock()
+
+	t.Run("validation of "+string(runtime.FeatureContainerPortAllocation), func(t *testing.T) {
+		err := runtime.ParseFeatures(string(runtime.FeatureContainerPortAllocation) + "=true")
+		assert.NoError(t, err)
+
+		// // container was not found
```

Comment:
Nit: double comment sign.
```suggestion
		// container was not found
```